src/WebJobs.Script.WebHost/Security/SecretsUtility.cs

```diff
@@ -17,5 +18,71 @@ public static string GetNonDecryptableName(string secretsPath)
             }
             return secretsPath + $".{ScriptConstants.Snapshot}.{timeStamp}.json";
         }
+
+        public static bool TryGetEncryptionKey(out byte[] key, IEnvironment environment = null)
```

Comment:
For our two token auth mechanisms, you can see that we were using separate logic to get the key. I want to centralize this logic to discrepancies don't crop up. As you can see, we already had different resolution logic for the key in the two token paths.